govcd/api.go

```diff
@@ -27,10 +27,7 @@ type Client struct {
 	Http          http.Client // HttpClient is the client to use. Default will be used if not provided.
 }
 
-// NewRequest creates a new HTTP request and applies necessary auth headers if
-// set.
-func (cli *Client) NewRequest(params map[string]string, method string, reqUrl url.URL, body io.Reader) *http.Request {
-
+func (cli *Client) NewRequestWitNotEncodedParams(params map[string]string, notEncodeParams map[string]string, method string, reqUrl url.URL, body io.Reader) *http.Request {
```

Comment:
What's the need for `NewRequestWithNotEndodedParams`? I think I could guess, but a detailed commend on why it is here and when to use it in preference to `NewRequest` should be needed.
If I guess correctly the need for this function, then the argument `notEncodeParams` should be `notEncodedParams`.

Reply:
Added comment.